In this step, we import the essential Python libraries used for data manipulation, visualisation, and machine learning throughout the project.

- pandas:  Data manipulation and analysis

- numpy: Numerical operations

- matplotlib and seaborn: Data visualisation

In [2]:
import pandas as pd
import numpy as np # for mathematical operations
import seaborn as sns # for data visualization
import matplotlib.pyplot as plt # for plotting graphs
%matplotlib inline
import warnings
warnings.filterwarnings('ignore') # to ignore warnings

For this problem, three CSV files are provided: train, test, and sample submission.

- Train File: Used to train the model. It contains all the independent variables (features) as well as the target variable.

- Test File: Contains only the independent variables. The trained model will be applied to this dataset to predict the target variable.

- Sample Submission File: Provides the required format for submitting the model’s predictions.

In [6]:
train_data = pd.read_csv('/Users/hamza/Desktop/projects/Loan_Prediction/Loan_Prediction/datasets/train_ctrUa4K.csv')
test_data = pd.read_csv('/Users/hamza/Desktop/projects/Loan_Prediction/Loan_Prediction/datasets/test_lAUu6dG.csv')

# Create a copy of the original data
train_data_copy = train_data.copy()
test_data_copy = test_data.copy()

This section we have a look at the structure of the train and test datasets.

In [16]:
train_data.columns, test_data.columns


(Index(['Loan_ID', 'Gender', 'Married', 'Dependents', 'Education',
        'Self_Employed', 'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
        'Loan_Amount_Term', 'Credit_History', 'Property_Area', 'Loan_Status'],
       dtype='object'),
 Index(['Loan_ID', 'Gender', 'Married', 'Dependents', 'Education',
        'Self_Employed', 'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
        'Loan_Amount_Term', 'Credit_History', 'Property_Area'],
       dtype='object'))

'Loan_Status' feature is missing form the test data. We will predict the 'Loan_Status' using the model built using the train data.

Print data types for each variable

In [12]:
train_data.dtypes

Loan_ID               object
Gender                object
Married               object
Dependents            object
Education             object
Self_Employed         object
ApplicantIncome        int64
CoapplicantIncome    float64
LoanAmount           float64
Loan_Amount_Term     float64
Credit_History       float64
Property_Area         object
Loan_Status           object
dtype: object

There are three main data types present in this dataset:

- Object: Represents categorical variables stored as text (e.g., gender, education level).
- int64: Represents integer numerical variables (whole numbers).
- float64: Represents numerical variables containing decimal values.

Both int64 and float64 are numerical data types, while object typically corresponds to categorical features.

In [15]:
train_data.shape,  test_data.shape

((614, 13), (367, 12))

There are 614 rows and 13 columns in the train dataset, and 367 rows and 12 columns in the test dataset.